docs/cudf/source/developer_guide/ARCHITECTURE.md

```diff
@@ -0,0 +1,208 @@
+# cuDF Architecture
+
+The cuDF library is a GPU-accelerated, [Pandas-like](https://pandas.pydata.org/) DataFrame library.
+pandas APIs provide users a greate deal of power and flexibility, and we aim to match that.
+As a result, a key design challenge for cuDF is finding the simplest, most performant approaches to mimic pandas APIs.
+
+At a high level, cuDF is structured in three layers, each of which serves a distinct purpose in this regard:
+
+1. The Frame layer: The user-facing implementation of pandas-like data structures.
+2. The Column layer: The core internal data structures used to bridge the gap to our lower-level implementations.
+3. The Cython layer: The wrappers around the fast C++ `libcudf` library.
+
+In this document we will review each of these layers, their roles, and the requisite tradeoffs.
+
+
+## The Frame layer
+
+Broadly speaking, the `Frame` layer is composed of two types of objects: indexed tables and indexes.
+The mapping between these types and cuDF data types is not obvious, however.
```

Comment:
Does "data type" here correspond to "dtypes", or the user-facing cuDF types?

If the latter, maybe we can say instead:

```suggestion
The mapping between these types and user-facing types like `Series` and `DataFrame` is not obvious, however.
```

Reply:
Let me reword a little more thoroughly. I don't think this sentence is conveying what I originally wanted it to.

Reply:
Actually on a second reading, I think we can get rid of these three sentences entirely. I don't think they are saying anything useful.